### PR TITLE
fixes #2795 - adding thread dump via -TTIN signal

### PR DESCRIPTION
**Pull request by lzap**
SSIA See http://projects.theforeman.org/issues/2795

### CODE REVIEW

**Comment by dmitri-d** [comment]
nice.

File: config/initializers/sig_ttin_trap.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# This initalizer overrides the TTIN signal:
+#   http://www.gnu.org/software/coreutils/manual/html_node/Signal-specifications.html
+#   TTIN - Background process attempting read.
+# This signal is not used with the thin server so we can override and add our additional
+# behavior.  if/when we switch to Passenger we can use their existing ABRT (SIGABRT) trap
+# which prints a threaddump and backtrace
```
Comment (ekohl) [inline config/initializers/sig_ttin_trap.rb]: Foreman is already deployed using passenger in the default installer.

Comment (lzap) [inline config/initializers/sig_ttin_trap.rb]: Yeah right, I tested that and it works apparently.

```
# kill -SIGABRT pid

[ 2013-07-16 12:07:29.8258 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr] [ 2013-07-16 12:07:29.8254 7945/0x000000023ca0f8(Main thread) request_handler.rb:389 ]: ========== Process 7945: backtrace dump ==========
[ 2013-07-16 12:07:29.8259 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr] ------------------------------------------------------------
[ 2013-07-16 12:07:29.8259 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr] # Thread: #<Thread:0x000000023ca0f8 run>(Main thread), [main thread], [current thread], alive = true
[ 2013-07-16 12:07:29.8259 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr] ------------------------------------------------------------
[ 2013-07-16 12:07:29.8259 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/lib/phusion_passenger/utils.rb:149:in `block in global_backtrace_report'
[ 2013-07-16 12:07:29.8260 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/lib/phusion_passenger/utils.rb:148:in `each'
[ 2013-07-16 12:07:29.8260 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/lib/phusion_passenger/utils.rb:148:in `global_backtrace_report'
[ 2013-07-16 12:07:29.8260 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/lib/phusion_passenger/request_handler.rb:389:in `print_status_report'
[ 2013-07-16 12:07:29.8260 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/lib/phusion_passenger/request_handler.rb:375:in `block in install_useful_signal_handlers'
[ 2013-07-16 12:07:29.8260 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/lib/phusion_passenger/request_handler.rb:510:in `call'
[ 2013-07-16 12:07:29.8260 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/lib/phusion_passenger/request_handler.rb:510:in `select'
[ 2013-07-16 12:07:29.8260 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/lib/phusion_passenger/request_handler.rb:510:in `wait_until_termination_requested'
[ 2013-07-16 12:07:29.8261 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/lib/phusion_passenger/request_handler.rb:201:in `main_loop'
[ 2013-07-16 12:07:29.8261 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/helper-scripts/rack-preloader.rb:130:in `<module:App>'
[ 2013-07-16 12:07:29.8261 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/helper-scripts/rack-preloader.rb:6:in `<module:PhusionPassenger>'
[ 2013-07-16 12:07:29.8261 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/helper-scripts/rack-preloader.rb:5:in `<main>'
[ 2013-07-16 12:07:29.8261 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr] 
[ 2013-07-16 12:07:29.8261 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr] ------------------------------------------------------------
[ 2013-07-16 12:07:29.8261 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr] # Thread: #<Thread:0x00000008ae5a58 sleep>(Worker 1), alive = true
[ 2013-07-16 12:07:29.8261 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr] ------------------------------------------------------------
[ 2013-07-16 12:07:29.8262 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/lib/phusion_passenger/request_handler/thread_handler.rb:125:in `accept_and_process_next_request'
[ 2013-07-16 12:07:29.8262 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/lib/phusion_passenger/request_handler/thread_handler.rb:108:in `main_loop'
[ 2013-07-16 12:07:29.8262 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/lib/phusion_passenger/request_handler.rb:441:in `block (3 levels) in start_threads'
[ 2013-07-16 12:07:29.8262 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr] 
[ 2013-07-16 12:07:29.8262 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr] ------------------------------------------------------------
[ 2013-07-16 12:07:29.8262 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr] # Thread: #<Thread:0x00000008ae5be8 sleep>(HTTP helper worker), alive = true
[ 2013-07-16 12:07:29.8262 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr] ------------------------------------------------------------
[ 2013-07-16 12:07:29.8262 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/lib/phusion_passenger/request_handler/thread_handler.rb:125:in `accept_and_process_next_request'
[ 2013-07-16 12:07:29.8262 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/lib/phusion_passenger/request_handler/thread_handler.rb:108:in `main_loop'
[ 2013-07-16 12:07:29.8263 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr]     /usr/lib/ruby/gems/1.8/gems/passenger-4.0.5/lib/phusion_passenger/request_handler.rb:457:in `block (2 levels) in start_threads'
[ 2013-07-16 12:07:29.8263 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr] 
[ 2013-07-16 12:07:29.8263 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr] 
[ 2013-07-16 12:07:29.8263 924/7fae35152700 Pool2/Implementation.cpp:1174 ]: [App 7891 stderr] [ 2013-07-16 12:07:29.8257 7945/0x000000023ca0f8(Main thread) request_handler.rb:390 ]: Threads: [#<Thread:0x00000008ae5a58 sleep>, #<Thread:0x00000008ae5be8 sleep>]
```

Comment (ekohl) [inline config/initializers/sig_ttin_trap.rb]: I didn't mean this shouldn't be added at all, just that the comment is inaccurate and should at least reflect the reality.

**Comment by mccun934** [comment]
For thin deployments it still seems useful to have this in.

**Comment by lzap** [comment]
Ok closing this one, I accidentaly deleted my branch from remote.

**Pull request closed without merging**